src/Api/WebSocket.hs

```diff
@@ -86,11 +89,13 @@ getDatumsByHashes hashes = do
       Left resp
     Left DatabaseErrorNotFound ->
       Right $ mkGetDatumsByHashesResponse Nothing
-    Right datums -> do
-      let datums' =
-            Vector.toList $
-              Vector.map (Aeson.toJSON . uncurry GetDatumsByHashesDatum) datums
-      Right $ mkGetDatumsByHashesResponse (Just datums')
+    Right datumsOrErrors ->
+      let rightDatums :: [(DataHash, PlutusData.Data)]
+          (_, rightDatums) =
+            bimap Map.toList Map.toList $ Map.mapEither id datumsOrErrors
+          datums' =
+            Aeson.toJSON <$> (uncurry GetDatumsByHashesDatum <$> rightDatums)
+       in Right $ mkGetDatumsByHashesResponse (Just datums')
```

Comment:
Shouldn't format of response also be changed to return which datums failed?

Reply:
That's the thing, I assume we didn't want to change the response type but we want to avoid the function failing if there is a decode error in just one of the datums.

Reply:
OK, let's merge it without breaking changes in public facing API, and create an issue to change return responses with CHANGELOG